src/main/java/org/generationcp/middleware/api/location/LocationService.java

```diff
@@ -36,4 +36,8 @@
 	 * @return the favorite project location ids
 	 */
 	List<Integer> getFavoriteProjectLocationIds(String programUUID);
+
+	long countLocations(LocationSearchRequest locationSearchRequest);
+
+	List<org.generationcp.middleware.api.location.Location> getLocations(LocationSearchRequest locationSearchRequest, Pageable pageable);
```

Comment:
Please check if we can refactor existing `getFilteredLocations` to return `List<org.generationcp.middleware.api.location.Location>`